src/components/Context/createComponent.js

```diff
@@ -1,39 +1,54 @@
-    const context = Object.keys(configuredContexts).reduce((memo, key) => {
-      memo = { ...memo, ...configuredContexts[key]() }; // eslint-disable-line no-param-reassign
-      return memo;
-    }, {});
-    payload.addContext(context);
-  };
+/*
+Copyright 2019 Adobe. All rights reserved.
+This file is licensed to you under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License. You may obtain a copy
+of the License at http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software distributed under
+the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+OF ANY KIND, either express or implied. See the License for the specific language
+governing permissions and limitations under the License.
+*/
+
+export default (config, logger, availableContexts, defaultContextNames) => {
+  let configuredContexts;
   return {
     namespace: "Context",
     lifecycle: {
       onComponentsRegistered() {
-        if (!config.context) {
-          logger.log(`No configured context.  Using default context.`);
-          configuredContexts = defaultContexts;
-          return;
-        }
-        if (!Array.isArray(config.context)) {
-          logger.warn(
-            `Invalid configured context.  Please specify an array of strings.`
-          );
-          configuredContexts = {};
-          return;
-        }
-        configuredContexts = config.context.reduce((memo, context) => {
-          if (availableContexts[context]) {
-            memo[context] = availableContexts[context]; // eslint-disable-line no-param-reassign
+        let configuredContextNames;
+
+        if (config.context) {
+          if (Array.isArray(config.context)) {
+            configuredContextNames = config.context;
           } else {
-            logger.warn(`Configured context ${context} is not available.`);
+            logger.warn(
+              `Invalid configured context. Please specify an array of strings.`
+            );
+            configuredContextNames = [];
           }
-          return memo;
-        }, {});
+        } else {
+          logger.log(`No configured context. Using default context.`);
+          configuredContextNames = defaultContextNames;
+        }
+
+        configuredContexts = configuredContextNames
+          .filter(configuredContextName => {
+            if (!availableContexts[configuredContextName]) {
+              logger.warn(
+                `Configured context ${configuredContextName} is not available.`
+              );
+              return false;
+            }
+            return true;
+          })
+          .map(
+            configuredContextName => availableContexts[configuredContextName]
+          );
       },
-      onBeforeEvent: onBeforeRequest,
-      onBeforeViewStart: onBeforeRequest
+      onBeforeEvent(event) {
+        configuredContexts.forEach(context => context(event));
```

Comment:
Note that context information was moved in the spec from `payload.context.contextName` to `payload.events[index].contextName`. Previously were building up the context object and then adding it to payload, but now we're passing `event` into each context provider and the context provider uses the requisite method on `event` to add pertinent data to the `event` object.